test/test.hs

```diff
@@ -162,6 +162,79 @@ macroTests = testGroup "macroTests" $
          snd (last (unwindMacros (bindInput [("a", "'xy'v")] coinput)
                                  (stringToKeyMacro "'aa'V")))
          @?= "xyyxyyxyyxyy"
+     , testCase "RepeatLast test 24" $
+         snd (last (unwindMacros (bindInput [("a", "'xy'vv")] coinput)
+                                 (stringToKeyMacro "'aa'vv")))
+         @?= "xyyyxyyyxyyyxyyy"
+     , testCase "RepeatLast test 25" $
+         snd (last (unwindMacros (bindInput [("a", "'xyv'v")] coinput)
+                                 (stringToKeyMacro "'a'a'vv'")))
+         @?= "xyyyxyyyxyyyxyyy"
+     -- , testCase "RepeatLast test 26" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+     --                                        , ("b", "'za'v")
+     --                                        , ("c", "'ab'v") ] coinput)
+     --                             (stringToKeyMacro "'c'v")))
+     --     @?= "xyyzxyyxyyzxyyxyyxyyzxyyxyyzxyyxyy"
+     -- , testCase "RepeatLast test 27" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'V")
+     --                                        , ("b", "'za'v")
+     --                                        , ("c", "'ab'v") ] coinput)
+     --                             (stringToKeyMacro "'c'v")))
+     --     @?= "xyxyzxyxyxyxyzxyxyxyxyxyxyzxyxyxyxyzxyxyxyxy"
+     -- , testCase "RepeatLast test 28" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+     --                                        , ("b", "'za'V")
+     --                                        , ("c", "'ab'v") ] coinput)
+     --                             (stringToKeyMacro "'c'v")))
+     --     @?= "xyyzxyyzxyyzxyyzxyyxyyzxyyzxyyzxyyzxyy"
+     -- , testCase "RepeatLast test 29" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+     --                                        , ("b", "'za'V")
+     --                                        , ("c", "'ab'V") ] coinput)
+     --                             (stringToKeyMacro "'c'v")))
+     --     @?= "xyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyy"
+     -- , testCase "RepeatLast test 30" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+     --                                        , ("b", "'za'V")
+     --                                        , ("c", "'ab'V") ] coinput)
+     --                             (stringToKeyMacro "'c'V")))
+     --     @?= "xyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyy"
+     -- , testCase "RepeatLast test 31" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+     --                                        , ("b", "'za'v")
+     --                                        , ("c", "'ab'V") ] coinput)
+     --                             (stringToKeyMacro "'c'V")))
+     --     @?= "xyyzxyyxyyzxyyxyyzxyyxyyzxyy"
```

Comment:
Shouldn't this one be `xyyzxyyxyyxyyzxyyxyyxyyzxyyxyyxyyzxyyxyy`?

Reply:
It should.

Reply:
That was a test. :D

Reply:
Darn, should've known better! Laughing hard right now. Well, at least the new implementation fails exactly when the old one does, so there's a strong chance they're doing the same thing :-D

Reply:
No, seriously, I'm a fake, I let my cat enter the test results. And I don't even have a cat.